src/shogun/evaluation/CrossValidationWrapper.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ *
+ */
+
+#ifndef CROSSVALIDATATION_WRAPPER_H__
+#define CROSSVALIDATATION_WRAPPER_H__
+
+#include <vector>
+#include <shogun/machine/Machine.h>
+#include <type_traits>
+#include <string_view>
+#include <memory>
+#include <queue>
+#include <utility>
+#include <shogun/evaluation/CrossValidation.h>
+
+#define IGNORE_IN_CLASSLIST
+
+namespace shogun
+{
+    template<typename MachineType>
+    class CrossValidationWrapper : public SGObject
+    {   
+    public:
+        CrossValidationWrapper(std::vector<std::pair<std::string_view, std::vector<float64_t>>> params, 
+            std::shared_ptr<CrossValidation> cross_validater): 
+            m_params(std::move(params)), m_cross_validater(cross_validater)
+        {
+        }
+
+        void fit(const std::shared_ptr<Features>& data, const std::shared_ptr<Labels>& labs)
+        {
+            compute_params_pair();
+            m_machine = m_cross_validater->get_machine();
+            std::priority_queue<std::pair<float64_t, 
+                std::vector<std::pair<std::string_view, float64_t>>>> best_param_candidates;
+            for(auto&& param_pair: std::as_const(m_params_pair))
+            {
+                for(auto&& [param_name, param_value]: std::as_const(param_pair))
+                {
+                    m_machine->put(param_name, param_value);
```

Comment:
I'd prefer if every parameter would act on a clone of the machine (without the data). This would make it easier to parallelize this code later

Reply:
this might kill using the queue, but we could just sort the results afterwards and each fold writes to an array